### PR TITLE
Datasource instance TTL

### PR DESCRIPTION
**Pull request by anniezhang77**
risk: low

### CODE REVIEW

File: acdc-core/src/main/scala/com/salesforce/mce/acdc/db/DatasetInstanceQuery.scala [hunk 1]
```diff
@@ -66,6 +66,10 @@ object DatasetInstanceQuery {
       .result
   }
 
+  def expire() = {
+    DatasetInstanceTable().filter(_.updatedAt < LocalDateTime.now().minusDays(90)).delete
```
Comment (realstraw) [inline acdc-core/src/main/scala/com/salesforce/mce/acdc/db/DatasetInstanceQuery.scala:70]: Instead of 90, let's make the days configurable

Comment (anniezhang77) [reply acdc-core/src/main/scala/com/salesforce/mce/acdc/db/DatasetInstanceQuery.scala:70]: as environment variable or in conf file is enough?

Comment (realstraw) [reply acdc-core/src/main/scala/com/salesforce/mce/acdc/db/DatasetInstanceQuery.scala:70]: If you put it as config, it would work with env too.

Comment (anniezhang77) [reply acdc-core/src/main/scala/com/salesforce/mce/acdc/db/DatasetInstanceQuery.scala]: like this https://github.com/salesforce/acdc/pull/42/commits/2395f95a43bbef9bc178f6803459dc708729ff86?

File: acdc-core/src/main/scala/com/salesforce/mce/acdc/db/DatasetInstanceQuery.scala [hunk 2]
```diff
@@ -66,6 +66,10 @@ object DatasetInstanceQuery {
       .result
   }
 
+  def expire(ttl: Int) = {
```
Comment (realstraw) [inline acdc-core/src/main/scala/com/salesforce/mce/acdc/db/DatasetInstanceQuery.scala:69]: let's rename `ttl` to `ttlDays` to make it clearer

File: acdc-ws/app/tasks/DataInstExpirationTask.scala [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+class DataInstExpirationTask @Inject()(
+  actorSystem: ActorSystem,
+  dbService: DatabaseService,
+  dbConfig: DbConfig
+)(implicit
+  ec: ExecutionContext
+) extends Logging {
+
+  val refreshDelay: FiniteDuration = 12.hour
+  val hostName = (InetAddress.getLocalHost.getHostName + UUID.randomUUID().toString).take(256)
```
Comment (realstraw) [inline acdc-ws/app/tasks/DataInstExpirationTask.scala:33]: let's start with the `randomUUID` and also have a separator between them to make sure we are not truncating that.

Comment (anniezhang77) [reply acdc-ws/app/tasks/DataInstExpirationTask.scala:33]: so truncating hostname instead?

**Review by realstraw — APPROVED** [review]

**Pull request merged**